exp/services/ledgerexporter/internal/resumablemanager.go

```diff
@@ -0,0 +1,138 @@
+	//    2. datastore had partial data in the requested range: absentLedger={a value > start and <= end}, ok=true
+	//    3. datastore had all data in the requested range: absentLedger=0, ok=false
+	FindStart(ctx context.Context, start, end uint32) (absentLedger uint32, ok bool, err error)
+}
+
+type resumableManagerService struct {
+	network           string
+	ledgerBatchConfig LedgerBatchConfig
+	dataStore         DataStore
+	archive           historyarchive.ArchiveInterface
+}
+
+func NewResumableManager(dataStore DataStore,
+	network string,
+	ledgerBatchConfig LedgerBatchConfig,
+	archive historyarchive.ArchiveInterface) ResumableManager {
+	return &resumableManagerService{
+		ledgerBatchConfig: ledgerBatchConfig,
+		network:           network,
+		dataStore:         dataStore,
+		archive:           archive,
+	}
+}
+
+func (rm resumableManagerService) FindStart(ctx context.Context, start, end uint32) (absentLedger uint32, ok bool, err error) {
+	if start < 1 {
+		return 0, false, errors.New("Invalid start value, must be greater than zero")
+	}
+
+	log := logger.WithField("start", start).WithField("end", end).WithField("network", rm.network)
+
+	networkLatest := uint32(0)
+	if end < 1 {
+		var latestErr error
+		networkLatest, latestErr = getLatestLedgerSequenceFromHistoryArchives(rm.archive)
+		if latestErr != nil {
+			err := errors.Wrap(latestErr, "Resumability of requested export ledger range, was not able to get latest ledger from network")
+			return 0, false, err
+		}
+		networkLatest = networkLatest + (getHistoryArchivesCheckPointFrequency() * 2)
+		logger.Infof("Resumability computed effective latest network ledger including padding of checkpoint frequency to be %d + for network=%v", networkLatest, rm.network)
+
+		if start > networkLatest {
+			// requested to start at a point beyond the latest network, resume not applicable.
+			return 0, false, errors.Errorf("Invalid start value of %v, it is greater than network's latest ledger of %v", start, networkLatest)
+		}
+	}
+
+	binarySearchStop := max(end, networkLatest)
+	binarySearchStart := start
+
+	log.Infof("Resumability is searching datastore for next absent object key of requested export ledger range")
+
+	rangeSize := max(int(binarySearchStop-binarySearchStart), 1)
+	var binarySearchError error
+	lowestAbsentIndex := sort.Search(rangeSize, binarySearchCallbackFn(&rm, ctx, binarySearchStart, binarySearchStop, &binarySearchError))
+	if binarySearchError != nil {
+		return 0, false, binarySearchError
+	}
+
+	if lowestAbsentIndex < 1 {
+		// data store had no data within search range
+		return start, true, nil
+	}
+
+	if lowestAbsentIndex < int(rangeSize) {
+		nearestAbsentLedgerSequence := binarySearchStart + uint32(lowestAbsentIndex)
+		log.Infof("Resumability determined next absent object start key of %d for requested export ledger range", nearestAbsentLedgerSequence)
+		return nearestAbsentLedgerSequence, true, nil
+	}
```

Comment:
```suggestion
	if lowestAbsentIndex < rangeSize {
		nearestAbsentLedgerSequence := binarySearchStart + uint32(lowestAbsentIndex)
		log.Infof("Resumability determined next absent object start key of %d for requested export ledger range", nearestAbsentLedgerSequence)
		return nearestAbsentLedgerSequence, true, nil
	}
```

no need to handle the `lowestAbsentIndex < 1` case separately. also, I think `rangeSize` is already an int so no need to cast it

Reply:
good catch, cleaned it up more - https://github.com/stellar/go/pull/5264/commits/a60db1963e7ccc03e495da1e99581dd0424499da